### PR TITLE
Remove track numbers from YouTube title

### PR DESCRIPTION
**Pull request by stephanebruckert**
**Describe the changes you made**
Solves https://github.com/web-scrobbler/web-scrobbler/issues/2272

**Additional context**
Regex used: https://regex101.com/r/gHh2TB/5

### CODE REVIEW

**Comment by alexesprit** [comment]
Could you add test cases where a video title should not be changed?

File: tests/content/util.js [hunk 1]
```diff
@@ -244,6 +244,18 @@ const PROCESS_YT_VIDEO_TITLE_DATA = [{
 	description: 'should remove 【genre】 from the beginning of the title',
 	args: ['【Genre】 Artist - Track'],
 	expected: { artist: 'Artist', track: 'Track' },
+}, {
+	description: 'should remove CD track number from the beginning of the title',
+	args: [' 01. Artist - Track'],
```
Comment (alexesprit) [inline tests/content/util.js:249]: Is the leading space added intentionally?

File: src/core/content/util.js [hunk 2]
```diff
@@ -567,7 +567,10 @@ const Util = {
 		}
 
 		// Remove [genre] or 【genre】 from the beginning of the title
-		const title = videoTitle.replace(/^((\[[^\]]+])|(【[^】]+】))\s*-*\s*/i, '');
+		let title = videoTitle.replace(/^((\[[^\]]+])|(【[^】]+】))\s*-*\s*/i, '');
+
+		// Remove track (CD and vinyl) numbers from the beginning of the title
+		title = title.replace(/^\s*((([a-zA-Z]{1,2})|([0-9]{1,2}))[1-9]?\.\s+)/i, '');
```
Comment (inverse) [inline src/core/content/util.js:573]: Would this simplify things? why so many braces? 

```suggestion
		title = title.replace(/^\s*([a-zA-Z]{1,2}|([0-9]{1,2})[1-9]?\.\s+/i, '');
```

File: tests/content/util.js [hunk 3]
```diff
@@ -244,6 +244,18 @@ const PROCESS_YT_VIDEO_TITLE_DATA = [{
...
+	expected: { artist: 'Artist', track: 'Track' },
+}, {
+	description: 'should remove vinyl track number from the beginning of the title',
+	args: [' A1. Artist - Track'],
+	expected: { artist: 'Artist', track: 'Track' },
+}, {
+	description: 'should remove complex vinyl track number from the beginning of the title',
+	args: [' AA1. Artist - Track'],
+	expected: { artist: 'Artist', track: 'Track' },
```
Comment (inverse) [inline tests/content/util.js:258]: Should we extend this with the ones that should be left intact? e.g. some of the test cases in the regex101 URL? I feel like this test suite should get a bit more extensive.

Comment (stephanebruckert) [reply tests/content/util.js:258]: I can add more tests, is it fine if I add them here or should I create a new function in the code as well as a new test function?

File: tests/content/util.js (hunk 1 above)
Comment (stephanebruckert) [reply tests/content/util.js:249]: Yes it was intentional, I'm adding more tests so I will cover space and no space

File: tests/content/util.js (hunk 3 above)
Comment (stephanebruckert) [reply tests/content/util.js:258]: Added more tests

**Comment by stephanebruckert** [comment]
@alexesprit @inverse, thanks for the review, I applied your feedback

**Comment by alexesprit** [comment]
@inverse Have you any suggestions?

**Comment by stephanebruckert** [comment]
@alexesprit I would love a patch release for this if you get a chance  :pray:

**Comment by alexesprit** [comment]
I think I will publish v2.33.0 instead.

**Comment by stephanebruckert** [comment]
@alexesprit nice one, thanks

**Pull request merged**